ComputePipeline/src/ComputePipeline.jl

```diff
@@ -194,10 +192,62 @@ struct ComputeGraph
     inputs::Dict{Symbol,Input}
     outputs::Dict{Symbol,Computed}
     onchange::Observable{Nothing}
+    observables::Dict{Symbol,Observable}
+end
+
+function get_observable!(attr::ComputeGraph, key::Symbol)
+    return get!(attr.observables, key) do
+        val = attr.outputs[key]
+        result = Observable(val[]; ignore_equal_values=true)
+        on(attr.onchange) do _
+            if isdirty(val)
+                _val = val[]
+                if !is_same(result[], _val)
+                    result[] = _val
+                end
+            end
+        end
```

Comment:
I don't think `isdirty` is safe to do here.

Consider a Graph a -> b -> c, where we first register an observable for c, then for b. If a is updated, a, b and c will be marked dirty. Then onchange triggers the c observable, which pulls c, marking a, b, c as resolved/not dirty. Next we trigger the b observable which sees b as not dirty and thus doesn't update.